health/health.d/wmi.conf

```diff
@@ -137,3 +137,175 @@ component: Disk
     delay: down 15m multiplier 1.5 max 1h
      info: disk space utilization
        to: sysadmin
+
+### TCP
+# Source: https://learn.microsoft.com/en-us/troubleshoot/windows-server/networking/default-dynamic-port-range-tcpip-chang
+# Windows by default allows 16383 (Windows 2008 or newer) and 3976 (old versions) parallel ports.
+#
+# We are adopting conservative values.
+#
+ template: wmi_concurrent_tcp
+       on: wmi.tcp_conns_active
+    class: TCP
+     type: Windows
+component: IIS
+       os: *
+    hosts: *
+     calc: $ipv4 + $ipv6
+    units: connections
+    every: 10s
+     warn: $this > (($status >= $WARNING)  ? (2783) : (2385))
+     crit: $this > (($status == $CRITICAL) ? (3578) : (3180))
+     info: Number of ports used is reaching levels that won't accept more connections.
+       to: sysadmin
```

Comment:
I think this alarm is wrong, the `wmi.tcp_conns_active` chart tracks the rate of active (whatever "active" is) TCP connections. You can't use the rate to check "Number of ports used".

Reply:
I do not think so, because `TCP` connections need ports opened.

Reply:
> because TCP connections need ports opened.

Yes. Did I say they don't? Still, I don't see how the rate can be used here.

Reply:
Let me be more clear:

- `tcp_conns_active` is - _Number of times TCP connections have made a direct transition from the CLOSED state to the SYN-SENT state_.
- `calc: $ipv4 + $ipv6` is the number of ^^ in the last second.
- And you are using that value to check for _Number of ports used is reaching levels that won't accept more connections_?

For me, it looks wrong.

Reply:
@thiagoftsm Where do we get the thresholds for this alert from? They seem oddly specific 2783? 3578?

Reply:
@shyamvalsan these are percentage of valid ranges for old Windows versions.

Reply:
Everything is wrong - using not the number of established connections but the rate of new + assuming all of them were initiated by the server.

Reply:
it was removed.